### ISSUE TITLE
Records Package Notifications and Comments- Harms

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#3433]
* As a Ministry user and IAO analyst
* I want to be notified when a records package is ready to retrieve
* so that I do not have to wait on a loading screen for the job to complete.

**Assumptions & Scope**	
This is related to user stories #3421 and #3231
Both of those user stories should be considered as 'background jobs' - and so we need a notification for the user to know when the job is complete.

What is IN scope?
Notifications and comments for harms export

Notification should be received by any assignee or watcher, both ministry and IAO.

What is NOT in scope?
Notifications for response package (see #  ) sign-off (needs completion of sign-off user story)

**Acceptance Criteria**	

**Scenario 1: Ministry User Exporting Records for Harms - Notify Success**
* GIVEN a Ministry or IAO user has activated the export records from the records log to send for harms
* WHEN the job completes, and the PDF is ready
* THEN the user will receive a notification indicating that "Records for harms are ready for export"

**Scenario 2: Ministry User Exporting Records for Harms - Notify Fail**
* GIVEN a Ministry or IAO user has activated the export records from the records log to send for harms
* WHEN the job has failed
* THEN the user will receive a notification indicating that "Preparing records for harms export failed. Please Try Again"

**Scenario 3: Click harms notification**
* GIVEN a user has received a harms notification
* AND the user has the notification open in the Notification Log
* WHEN the user clicks on the notification
* THEN the user will be directed to the Records Log

**Scenario 4: Ministry User Exporting Records for Harms - Comment Success**
* GIVEN a Ministry user has activated the export records from the records log to send for harms
* WHEN the job completes, and the PDF is ready
* THEN a system generated comment will be posted indicating that "Records for harms are ready for export"

**Scenario 5: Ministry User Exporting Records for Harms - Comment Fail**
* GIVEN a Ministry user has activated the export records from the records log to send for harms
* WHEN the job has failed
* THEN a system generated comment will be posted indicating "Preparing records for harms export failed. Please Try Again"

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by m-prodan** [comment]
Related ticket to #3421 - I am choosing to split the notifications between the PDF for harms, and the records release package, as I suspect one can be done sooner than the other. 

Also - have talked about where/how the user will access these records? My gut says it should be retrieved from attachments? 

cc: @lmullane @KyEggleston @JHarrietha-AOT @abin-aot @liseandtea

**Comment by lmullane** [comment]
Put is some suggested edits in the AC, removing the reference to the Job.

I would lean to having the Export in the Records log, but open to discussion.

**Comment by m-prodan** [comment]
That makes sense, keep things consistent

**Comment by divyav-aot** [comment]
Hi @KyEggleston this is already pushed to test-marshal. We need to find out a way to test this scenario.

**Comment by lmullane** [comment]
@KyEggleston, yes, let's add an AC.
